lib/http/apps/dao.js

```diff
@@ -20,20 +23,48 @@ const app = express();
  * @apiUse InternalServerError
  */
 app.post(`/`, (req, res) => {
+  let spaceId = req.body.spaceId;
   let spaceName = req.body.spaceName;
   let descHash = req.body.descHash;
   let coinName = req.body.coinName;
   let coinSymbol = req.body.coinSymbol;
   let coinDecimals = req.body.coinDecimals;
 
   console.log(`New DAO: ${spaceName} ${descHash} ${coinName} ${coinSymbol} ${coinDecimals}  with root: ${res.locals.wallet.address}`);
-  res.locals.kit.newInstance({ spaceName, descHash, coinName, coinSymbol, coinDecimals, root: res.locals.wallet.address }, { gasLimit: 5000000 }).then((dao) => {
-    res.status(201).json(dao);
-  }).catch(e => {
-    console.log(e);
-    res.sendStatus(500);
-  });
 
+  res.locals.kit.newInstance({ spaceId, spaceName, descHash, coinName, coinSymbol, coinDecimals, root: res.locals.wallet.address }, { gasLimit: 5000000 });
```

Comment:
as mentioned below. I'd do the saving of the dao file here. (but still directly do the `res.sendSatus`